app/app.py

```diff
@@ -78,9 +87,17 @@ def find_recipe_in_url_window_algo_based():
     instructions = request.form['instructions'].lower() == "true"
     ingredients = request.form['ingredients'].lower() == "true"
     try:
+        if stored_recipes_dao.url_in_db(url):
+            return jsonify(stored_recipes_dao.get_recipe_from_db(url, ingredients, instructions))
         all_text, title = text_extractor.get_all_text_from_url(url=url)
         ingred_paragraph, instr_paragraph = window_key_word_based_algo.extract(ingredients, instructions, all_text)
-        return create_json_response(ingred_paragraph, instr_paragraph, title, url)
+        response = create_json_response(ingred_paragraph, instr_paragraph, title, url)
+        with ThreadPoolExecutor(max_workers=2) as executor:
+            future = executor.submit(stored_recipes_dao.add_recipe_to_db,
+                                     response=response)
+            future.result()
```

Comment:
this about what happens here, we are creating a new ThreadPoolExcutor each time there is a call, how would you fix it?

Reply:
I bet the flask itself has a solution for that.
the main idea is to manage ThreadPool outside of the scope of the "route"